disttask/framework/dispatcher/dispatcher.go

```diff
@@ -418,34 +417,26 @@ func (d *dispatcher) processNormalFlow(gTask *proto.Task) (err error) {
 	}
 
 	// Generate all available TiDB nodes for this global tasks.
-	serverNodes, err1 := GenerateSchedulerNodes(d.ctx)
+	serverNodes, err1 := handle.GetEligibleInstances(d.ctx, gTask)
 	if err1 != nil {
 		return err1
 	}
+	if len(serverNodes) == 0 {
+		return errors.New("no available TiDB node")
```

Comment:
We do it in GenerateSchedulerNodes [line449](https://github.com/pingcap/tidb/pull/43880/files#diff-2a828989b540648a0bdef3d984d7946ea4fcb3c18718a4047b320c712ee1b855R449). Could we remove it?

Reply:
this line is moved from `GetInstanceForSubtask` actually